src/Microsoft.ML.PCA/PcaTransformer.cs

```diff
@@ -596,6 +597,78 @@ private static void TransformFeatures(IExceptionContext ectx, in VBuffer<float>
 
                 dst = editor.Commit();
             }
+
+            public bool CanSaveOnnx(OnnxContext ctx) => true;
+
+            public void SaveAsOnnx(OnnxContext ctx)
+            {
+                Host.CheckValue(ctx, nameof(ctx));
+
+                for (int i = 0; i < _numColumns; i++)
+                {
+                    var colPair = _parent.ColumnPairs[i];
+                    var transformInfo = _parent._transformInfos[i];
+                    string inputColumnName = colPair.inputColumnName;
+                    string outputColumnName = colPair.outputColumnName;
+                    if (!ctx.ContainsColumn(inputColumnName))
+                    {
+                        ctx.RemoveColumn(colPair.outputColumnName, false);
+                        continue;
+                    }
+
+                    if (!SaveAsOnnxCore(ctx, i, ctx.GetVariableName(inputColumnName),
+                        ctx.AddIntermediateVariable(transformInfo.OutputType, outputColumnName)))
+                    {
+                        ctx.RemoveColumn(outputColumnName, true);
+                    }
+                }
+            }
+
+            private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariableName, string dstVariableName)
```

Comment:
`bool` - When does this method ever return `false`?

Reply:
It doesn't in this case. I have tried to preserve the pattern used across other transformers in their implementation of ISaveAsOnnx.

Reply:
That seems less than ideal, given that you now have dead code when calling this method.

You can still have a `SaveAsOnnxCore` method, but just make it return `void` and remove the `if` check above.

Reply:
Resolved with latest checkin